bench/load/basic/launchWorkflow.go

```diff
@@ -109,159 +89,161 @@ func launcherWorkflow(ctx workflow.Context, config lib.BasicTestConfig) (string,
-				switch err := err.(type) {
-				case *workflow.TimeoutError:
-					if err.TimeoutType() == shared.TimeoutTypeHeartbeat && err.HasDetails() {
-						if err := err.Details(&batchStartID); err == nil {
-							batchStartID++
-						}
-					}
-				}
-				result.SuccessCount += runResult.SuccessCount
-				result.FailedCount += runResult.FailedCount
-			}
-			doneCh.Send(ctx, "done")
-		})
-	}
-	for i := 0; i < config.RoutineCount; i++ {
-		doneCh.Receive(ctx, nil)
+		actInput := launcherActivityParams{
+			RoutineID: i,
+			Count:     workflowPerActivity,
+			Config:    config,
+		}
+		f := workflow.ExecuteActivity(
+			ctx,
+			LauncherLaunchActivityName,
+			actInput,
+		)
+		err := f.Get(ctx, nil)
+		if err != nil {
+			return "", err
+		}
 	}
 
-	workflowWaitTime := stressWorkflowStartToCloseTimeout
+	workflowWaitTime := defaultStressWorkflowStartToCloseTimeout
 	if config.ExecutionStartToCloseTimeoutInSeconds > 0 {
 		workflowWaitTime = time.Duration(config.ExecutionStartToCloseTimeoutInSeconds) * time.Second
 	}
-	if err := workflow.Sleep(ctx, workflowWaitTime+workflowWaitTimeBuffer); err != nil {
+	workflowWaitTime += workflowWaitTimeBuffer
+	logger.Info(fmt.Sprintf("%v stressWorkflows are launched, now waiting for %v ...", config.TotalLaunchCount, workflowWaitTime))
+	if err := workflow.Sleep(ctx, workflowWaitTime); err != nil {
 		return "", fmt.Errorf("launcher workflow sleep failed: %v", err)
 	}
 
+	maxTolerantFailure := int32(float64(config.TotalLaunchCount) * config.FailureThreshold)
 	actInput := verifyActivityParams{
-		FailedWorkflowCount: int64(float64(config.TotalLaunchCount) * config.FailureThreshold),
-		WorkflowStartTime:   startTime.UnixNano(),
+		WorkflowStartTime:    startTime.UnixNano(),
+		ListWorkflowPageSize: maxTolerantFailure + 10,
 	}
 	actOptions := workflow.ActivityOptions{
 		ScheduleToStartTimeout: time.Minute,
 		StartToCloseTimeout:    time.Minute,
 		RetryPolicy: &cadence.RetryPolicy{
-			InitialInterval:          time.Second,
-			BackoffCoefficient:       2,
-			MaximumAttempts:          5,
-			NonRetriableErrorReasons: []string{errReasonValidationFailed},
+			InitialInterval:    time.Second,
+			BackoffCoefficient: 2,
+			MaximumAttempts:    5,
 		},
 	}
+	var result verifyActivityResult
 	ctx = workflow.WithActivityOptions(ctx, actOptions)
 	if err := workflow.ExecuteActivity(
 		ctx,
 		LauncherVerifyActivityName,
 		actInput,
-	).Get(ctx, nil); err != nil {
+	).Get(ctx, &result); err != nil {
 		return "", err
 	}
-	return result.String(), nil
+	passed := (result.TimeoutCount + result.OpenCount + result.FailedCount) <= int(maxTolerantFailure)
+	finalResult := fmt.Sprintf("TEST PASSED: %v; Details report: timeoutCount: %v, failedCount: %v, openCount:%v, launchCount: %v, maxThreshold:%v",
+		passed, result.TimeoutCount, result.FailedCount, result.OpenCount, config.TotalLaunchCount, maxTolerantFailure)
+	return finalResult, nil
```

Comment:
I think we may still want the bench workflow to fail when the maxTolerantFailure count is exceeded so that operation can easily tell if the bench has passed or not. 

We also have another `cron` bench, which is responsible for running multiple other bench loads. That bench's implementation also relies on the individual bench to failed when verification is not passed.

Reply:
Sure, will make that happen 👍